### PR TITLE
Add initial update-dependencies script.

### PR DESCRIPTION
**Pull request by eerhardt**
This script can be used to automatically update dotnet/cli's
dependencies on other repos like CoreFX, NuGet, etc.

Currently the script supports updating the CoreFX dependencies.

@piotrpMSFT @livarcocc @anurse @Sridhar-MS

### CODE REVIEW

File: scripts/update-dependencies/UpdateFilesTargets.cs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        [Target(nameof(GetDependencies), nameof(ReplaceVersions))]
+        public static BuildTargetResult UpdateFiles(BuildTargetContext c) => c.Success();
+
+        /// <summary>
+        /// Gets all the dependency information and puts it in the build properties.
+        /// </summary>
+        [Target]
+        public static BuildTargetResult GetDependencies(BuildTargetContext c)
+        {
+            string coreFxLkgVersion = s_client.GetStringAsync("https://raw.githubusercontent.com/eerhardt/versions/master/corefx/release/1.0.0-rc2/LKG.txt").Result;
```
Comment (eerhardt) [inline scripts/update-dependencies/UpdateFilesTargets.cs]: Oops.  Missed this one that needs to be parameterized.  I'll update this in the morning to read this from an Environment Variable.

File: scripts/update-dependencies/project.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "NuGet.Versioning": "3.4.0-rtm-0764",
+    "Newtonsoft.Json": "7.0.1",
+    "Octokit": "0.18.0",
+    "Microsoft.Net.Http": "2.2.29"
+  },
+  "frameworks": {
+    "netstandardapp1.5": {
+      "imports": [
+        "dnxcore50",
+        "portable-net45"
```
Comment (anurse) [inline scripts/update-dependencies/project.json]: I think this has to be `portable-net45+win8`. If there's a package that has `portable-net45+[something else]` it is not guaranteed to work in `netstandardapp1.5`. Is there a package that falls into that bucket?

**Comment by anurse** [comment]
LGTM. Minor concern about the portable import but otherwise good.

File: scripts/update-dependencies/project.json (hunk 2 above)
Comment (eerhardt) [inline scripts/update-dependencies/project.json]: Yes, the package that falls into this bucket is:  https://www.nuget.org/packages/Octokit/0.18.0, which we are using to create the PR on GitHub.  It has "portable-net45+wp80+win+wpa81" from looking at the package.

It happens to be working both on my machine and in the VSO build. I guess I haven't tried it on a Unix machine... Do you think I should add the full portable string?

Comment (eerhardt) [inline scripts/update-dependencies/project.json]: Octokit also has "netcore451" as a target... FYI

Comment (anurse) [inline scripts/update-dependencies/project.json]: How about `portable-net45+win`. That should be enough. For what it's worth, I think we've engaged with Octokit to get it ported to `netstandard` (but like most people they're waiting for things to stabilize)

**Comment by eerhardt** [comment]
@dotnet-bot test this please

**Pull request merged**